third_party/forked/shell2junit/sh2ju.sh

```diff
@@ -101,37 +101,37 @@ function juLog() {
   # eval the command sending output to a file
   outf=/var/tmp/ju$$.txt
   errf=/var/tmp/ju$$-err.txt
-  >${outf}
+  :>${outf}
   echo ""                         | tee -a ${outf}
   echo "+++ Running case: ${class}.${name} " | tee -a ${outf}
-  echo "+++ working dir: "`pwd`           | tee -a ${outf}
+  echo "+++ working dir: $(pwd)"           | tee -a ${outf}
   echo "+++ command: ${cmd}"            | tee -a ${outf}
-  ini=`${date} +%s.%N`
+  ini="$(${date} +%s.%N)"
   # execute the command, temporarily swapping stderr and stdout so they can be tee'd to separate files,
   # then swapping them back again so that the streams are written correctly for the invoking process
   ( (eVal "${cmd}" | tee -a ${outf}) 3>&1 1>&2 2>&3 | tee ${errf}) 3>&1 1>&2 2>&3
-  evErr=`cat ${errfile}`
+  evErr="$(cat ${errfile})"
   rm -f ${errfile}
-  end=`${date} +%s.%N`
+  end="$(${date} +%s.%N)"
   echo "+++ exit code: ${evErr}"        | tee -a ${outf}
 
   # set the appropriate error, based in the exit code and the regex
   [[ ${evErr} != 0 ]] && err=1 || err=0
-  out=`cat $outf | ${SED} -e 's/^\([^+]\)/| \1/g'`
-  if [ ${err} = 0 -a -n "${ereg:-}" ]; then
-      H=`echo "${out}" | egrep ${icase} "${ereg}"`
+  out="$(${SED} -e 's/^\([^+]\)/| \1/g' "$outf")"
+  if [ ${err} = 0 ] && [ -n "${ereg:-}" ]; then
```

Comment:
For reference - [SC2166](https://github.com/koalaman/shellcheck/wiki/SC2166) - _"Prefer [ p ] && [ q ] as [ p -a q ] is not well defined."_